flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -80,7 +82,7 @@ protected JobExceptionsInfo handleRequest(HandlerRequest<EmptyRequestBody, JobEx
 
 	@Override
 	public Collection<ArchivedJson> archiveJsonWithPath(AccessExecutionGraph graph) throws IOException {
-		ResponseBody json = createJobExceptionsInfo(graph, MAX_NUMBER_EXCEPTION_TO_REPORT);
+		ResponseBody json = createJobExceptionsInfo(graph, Iterables.size(graph.getAllExecutionVertices()));
```

Comment:
This change is out-of-scope of this PR and is a bit dangerous since it can significantly change current behavior; for example if you have a massive deployment and all of them fail for some reason (like a class-loading issue).

I'd revert it for the time being.